dattri/algorithm/influence_function.py

```diff
@@ -123,14 +132,21 @@ def transformation_on_query(
             torch.Tensor: The transformation on the query. Normally it is a 2-d
                 dimensional tensor with the shape of (batchsize, transformed_dimension).
         """
-        from dattri.func.hessian import ihvp_arnoldi
-
-        self.ihvp_func = ihvp_arnoldi(
-            partial(self.task.get_target_func(), data_target_pair=train_data),
-            **transformation_kwargs,
-        )
-        model_params, _ = self.task.get_param(index)
-        return self.ihvp_func((model_params,), query).detach()
+        from dattri.func.projection import arnoldi_project
+
+        if self.arnoldi_projector is None:
+            feature_dim = query.shape[1]
+            func = partial(self.task.get_target_func(), data_target_pair=train_data)
```

Comment:
may need to use `get_loss_func` after the merge of renaming PR

Reply:
yeah, if renaming PR is merged first, I can modify this.

Reply:
fixed.